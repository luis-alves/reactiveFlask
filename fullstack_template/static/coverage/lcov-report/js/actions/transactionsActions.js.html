<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/actions/transactionsActions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">js/actions</a> transactionsActions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/41</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/28</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/41</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import axios from "axios"
&nbsp;
&nbsp;
export function <span class="fstat-no" title="function not covered" >fetchTransactions(</span>) {
<span class="cstat-no" title="statement not covered" >  return <span class="fstat-no" title="function not covered" >fu</span>nction (dispatch) {</span>
<span class="cstat-no" title="statement not covered" >    dispatch({type: "FETCH_TRANSACTIONS_FETCHING"})</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    axios.get(window.location.href + 'transactions')</span>
    .then(<span class="fstat-no" title="function not covered" >re</span>sponse =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({type: "FETCH_TRANSACTIONS_FULFILLED", payload: response.data})</span>
    })
    .catch(<span class="fstat-no" title="function not covered" >er</span>r =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({type: "FETCH_TRANSACTIONS_REJECTED", payload: err})</span>
    })
  }
}
/* Update reconcile data */
export function <span class="fstat-no" title="function not covered" >updateTransactions(</span>el) {
<span class="cstat-no" title="statement not covered" >  return <span class="fstat-no" title="function not covered" >fu</span>nction (dispatch) {</span>
<span class="cstat-no" title="statement not covered" >    dispatch({type: "UPDATE_TRANSACTIONS_FETCHING"})</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    axios.post(window.location.href + 'reconcileflag', {</span>
      elementsId: el
      // elementsId: e.target.parentElement.parentElement.id,
    }
  ).then(<span class="fstat-no" title="function not covered" >re</span>sponse =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({</span>
        type: "UPDATE_TRANSACTIONS_FULFILLED",
        payload: response.data.reconcile,
        id: response.data.id})
    })
    .catch(<span class="fstat-no" title="function not covered" >er</span>r =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({</span>
        type: "UPDATE_TRANSACTIONS_REJECTED",
        payload: err})
    })
    // Totals
    // Fetching cleared balance
<span class="cstat-no" title="statement not covered" >    dispatch({type: "FETCH_BALANCE_FETCHING"})</span>
<span class="cstat-no" title="statement not covered" >    axios.get(window.location.href + 'balance')</span>
    .then(<span class="fstat-no" title="function not covered" >re</span>sponse =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({type: "FETCH_BALANCE_FULFILLED", payload: response.data})</span>
    })
    .catch(<span class="fstat-no" title="function not covered" >er</span>r =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({type: "FETCH_BALANCE_REJECTED", payload: err})</span>
    })
  }
}
/* Update database of the color of the bookmarks */
export function <span class="fstat-no" title="function not covered" >updateTransactionsBookmark(</span>color, id) {
<span class="cstat-no" title="statement not covered" >  return <span class="fstat-no" title="function not covered" >fu</span>nction (dispatch) {</span>
<span class="cstat-no" title="statement not covered" >    dispatch({type: "UPDATE_TRANSACTIONS_FETCHING"})</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    axios.post(window.location.href + 'bookmark', {</span>
      color: color,
      id: id
      // elementsId: e.target.parentElement.parentElement.id,
      // elementsId: e.target.parentElement.parentElement.id,
    }
    ).then(<span class="fstat-no" title="function not covered" >re</span>sponse =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({</span>
        type: "UPDATE_TRANSACTIONS_FULFILLED",
        payload: response.data.bookmark,
        id: response.data.id})
    })
    .catch(<span class="fstat-no" title="function not covered" >er</span>r =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({</span>
        type: "UPDATE_TRANSACTIONS_REJECTED",
        payload: err})
    })
    // Fetch updated transactions
<span class="cstat-no" title="statement not covered" >    dispatch({type: "FETCH_TRANSACTIONS_FETCHING"})</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    axios.get(window.location.href + 'transactions')</span>
    .then(<span class="fstat-no" title="function not covered" >re</span>sponse =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({type: "FETCH_TRANSACTIONS_FULFILLED", payload: response.data})</span>
    })
    .catch(<span class="fstat-no" title="function not covered" >er</span>r =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({type: "FETCH_TRANSACTIONS_REJECTED", payload: err})</span>
    })
  }
}
&nbsp;
/* Update database with the data from input fields */
export function <span class="fstat-no" title="function not covered" >updateTransactionsInputData(</span>data, date) {
&nbsp;
<span class="cstat-no" title="statement not covered" >  return <span class="fstat-no" title="function not covered" >fu</span>nction (dispatch) {</span>
<span class="cstat-no" title="statement not covered" >    dispatch({type: "INPUT_TRANSACTION_FETCHING"})</span>
<span class="cstat-no" title="statement not covered" >    axios.post(window.location.href + 'input', {</span>
      payee: data.payee,
      date: date,
      category: data.category,
      memo: data.memo,
      outflow: data.outflow,
      inflow: data.inflow,
      id: data.id
    }
    ).then(<span class="fstat-no" title="function not covered" >re</span>sponse =&gt; {
      // console.log(response);
<span class="cstat-no" title="statement not covered" >      dispatch({</span>
        type: "INPUT_TRANSACTION_FULFILLED",
        payload: response.data,
        id: response.data.id})
    })
    .catch(<span class="fstat-no" title="function not covered" >er</span>r =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({</span>
        type: "INPUT_TRANSACTION_REJECTED",
        payload: err})
    })
    // Fetch updated transactions
<span class="cstat-no" title="statement not covered" >    dispatch({type: "FETCH_TRANSACTIONS_FETCHING"})</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    axios.get(window.location.href + 'transactions')</span>
    .then(<span class="fstat-no" title="function not covered" >re</span>sponse =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({type: "FETCH_TRANSACTIONS_FULFILLED", payload: response.data})</span>
    })
    .catch(<span class="fstat-no" title="function not covered" >er</span>r =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({type: "FETCH_TRANSACTIONS_REJECTED", payload: err})</span>
    })
  }
}
&nbsp;
/* Update database for the checkboxes */
export function <span class="fstat-no" title="function not covered" >updateTransactionsCheckbox(</span>id) {
<span class="cstat-no" title="statement not covered" >  return <span class="fstat-no" title="function not covered" >fu</span>nction (dispatch) {</span>
<span class="cstat-no" title="statement not covered" >    dispatch({type: "UPDATE_TRANSACTIONS_CHECKBOX_FETCHING"})</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    axios.post(window.location.href + 'checkbox', {</span>
      id: id
    }
    ).then(<span class="fstat-no" title="function not covered" >re</span>sponse =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({</span>
        type: "UPDATE_TRANSACTIONS_CHECKBOX_FULFILLED",
        payload: response.data.checkbox,
        id: response.data.id})
    })
    .catch(<span class="fstat-no" title="function not covered" >er</span>r =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({</span>
        type: "UPDATE_TRANSACTIONS_CHECKBOC_REJECTED",
        payload: err})
    })
    // Fetch updated transactions
<span class="cstat-no" title="statement not covered" >    dispatch({type: "FETCH_TRANSACTIONS_FETCHING"})</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    axios.get(window.location.href + 'transactions')</span>
    .then(<span class="fstat-no" title="function not covered" >re</span>sponse =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({type: "FETCH_TRANSACTIONS_FULFILLED", payload: response.data})</span>
    })
    .catch(<span class="fstat-no" title="function not covered" >er</span>r =&gt; {
<span class="cstat-no" title="statement not covered" >      dispatch({type: "FETCH_TRANSACTIONS_REJECTED", payload: err})</span>
    })
  }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Feb 22 2018 22:26:17 GMT+0100 (WAT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
