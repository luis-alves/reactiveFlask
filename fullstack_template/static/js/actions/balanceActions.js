
import axios from "axios"

export function fetchBalance() {
  return function (dispatch) {
    dispatch({type: "FETCH_BALANCE_FETCHING"})
    axios.get(window.location.href + 'balance')
    .then(response => {
      console.log(response.data);
      dispatch({type: "FETCH_BALANCE_FULFILLED", payload: response.data})
    })
    .catch(err => {
      dispatch({type: "FETCH_BALANCE_REJECTED", payload: err})
    })
  }
}
