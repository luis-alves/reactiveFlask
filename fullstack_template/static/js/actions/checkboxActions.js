import axios from "axios"


/* Remove all checkboxes ticks and enable only the one selected */
export function updateCheckbox(id) {
  return function (dispatch) {
    dispatch({type: "UPDATE_ALL_CHECKBOX_FETCHING"})

    axios.post(window.location.href + 'untick', {
      id: id
    })
    .then(response => {
      dispatch({
        type: "UPDATE_ALL_CHECKBOX_FULFILLED",
        payload: response.data.checkbox,
        id: response.data.id})
    })
    .catch(err => {
      dispatch({
        type: "UPDATE_ALL_CHECKBOX_REJECTED",
        payload: err})
    })
  }
}
