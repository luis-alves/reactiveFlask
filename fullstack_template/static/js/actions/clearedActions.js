
import axios from "axios"

export function fetchClearedBalance() {
  return function (dispatch) {
    dispatch({type: "FETCH_CLEARED_FETCHING"})
    axios.get(window.location.href + 'cleared_balance')
    .then(response => {
      dispatch({type: "FETCH_CLEARED_FULFILLED", payload: response.data})
    })
    .catch(err => {
      dispatch({type: "FETCH_CLEARED_REJECTED", payload: err})
    })
  }
}
