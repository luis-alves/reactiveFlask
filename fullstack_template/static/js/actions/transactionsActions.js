import axios from "axios"


export function fetchTransactions() {
  return function (dispatch) {
    dispatch({type: "FETCH_TRANSACTIONS_FETCHING"})

    axios.get(window.location.href + 'transactions')
    .then(response => {
      dispatch({type: "FETCH_TRANSACTIONS_FULFILLED", payload: response.data})
    })
    .catch(err => {
      dispatch({type: "FETCH_TRANSACTIONS_REJECTED", payload: err})
    })
  }
}

export function updateTransactions(e) {
  return function (dispatch) {
    dispatch({type: "UPDATE_TRANSACTIONS_FETCHING"})

    axios.post(window.location.href + 'reconcileflag', {
      elementsId: e.target.parentElement.parentElement.id,
      // elementsId: e.target.parentElement.parentElement.id,
    }
  ).then(response => {
      dispatch({
        type: "UPDATE_TRANSACTIONS_FULFILLED",
        payload: response.data.reconcile,
        id: response.data.id})
    })
    .catch(err => {
      dispatch({
        type: "UPDATE_TRANSACTIONS_REJECTED",
        payload: err})
    })
    // Totals
    // Fetching cleared balance
    dispatch({type: "FETCH_BALAMCE_FETCHING"})
    axios.get(window.location.href + 'balance')
    .then(response => {
      dispatch({type: "FETCH_BALANCE_FULFILLED", payload: response.data})
    })
    .catch(err => {
      dispatch({type: "FETCH_BALANCE_REJECTED", payload: err})
    })
  }
}
