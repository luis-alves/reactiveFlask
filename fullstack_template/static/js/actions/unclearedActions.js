
import axios from "axios"

export function fetchUnclearedBalance() {
  return function (dispatch) {
    dispatch({type: "FETCH_UNCLEARED_FETCHING"})
    axios.get(window.location.href + 'uncleared_balance')
    .then(response => {
      dispatch({type: "FETCH_UNCLEARED_FULFILLED", payload: response.data})
    })
    .catch(err => {
      dispach({type: "FETCH_UNCLEARED_REJECTED", payload: err})
    })
  }
}
