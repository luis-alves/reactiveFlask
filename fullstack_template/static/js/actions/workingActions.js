import axios from "axios"


export function fetchWorkingBalance() {
  return function (dispatch) {
    dispatch({type: "FETCH_WORKING_FETCHING"})
    axios.get(window.location.href + 'working_balance')
    .then(response => {
      dispatch({type: "FETCH_WORKING_FULFILLED", payload: response.data})
    })
    .catch(err => {
      dispach({type: "FETCH_WORKING_REJECTED", payload: err})
    })
  }
}
