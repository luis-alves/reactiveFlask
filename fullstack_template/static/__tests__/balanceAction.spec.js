import React from 'react'
import expect from 'expect'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import thunk from 'redux-thunk'
import fetchMock from 'fetch-mock'

import { fetchBalance } from '../js/actions/balanceActions'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

describe('check actions', () => {
  it('should have a type of CHANGE_LOCATION', function() {
    const store = mockStore({type: 'FETCH_BALANCE_FULFILLED', balance: 30})
    store.dispatch({type: 'FETCH_BALANCE_FULFILLED', balance: 30})
    console.log(store.getState())
    // return store.dispatch(fetchBalance())
    expect(store.dispatch(fetchBalance()).toMatchSnapshot())
    // return store.dispatch(fetchBalance()).then(() => {
      // expect(store.getActions()).toMatchSnapshot()
    // })
  });
})
