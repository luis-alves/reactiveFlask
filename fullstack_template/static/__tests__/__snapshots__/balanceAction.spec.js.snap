// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check actions should have a type of CHANGE_LOCATION 1`] = `
Object {
  "type": "FETCH_BALANCE_FULFILLED",
}
`;
